AsyncDisplayKit/ASMapNode.mm

```diff
@@ -14,11 +14,13 @@
 #import <AsyncDisplayKit/ASCenterLayoutSpec.h>
 #import <AsyncDisplayKit/ASThread.h>
 #import <AsyncDisplayKit/ASInternalHelpers.h>
+#import <AsyncDisplayKit/ASLayout.h>
 
 @interface ASMapNode()
 {
   ASDN::RecursiveMutex _propertyLock;
   MKMapSnapshotter *_snapshotter;
+  BOOL _snapshotNeeded;
```

Comment:
I'd rather not introduce another ivar to handle this, it introduces managed state, and I'd rather ASMapNode had constraints where it could figure out if it needed a new snapshot without manually having to set a flag.